### PR TITLE
fix: (temporarily) patch `flox publish`

### PR DESCRIPTION
**Pull request by limeytexan**
## Proposed Changes

This PR optimizes the `flox publish` flow to address a few specific high-value use cases. Most of the publish logic is due to be overhauled in the near term, so this should be viewed as a short-term repair while we wait for the ultimate fix.

## Current Behavior

`flox publish` currently prompts to reconfirm all parameters on every invocation unless they are explicitly provided as command-line options. This PR updates the flow to be more streamlined, prompting only when a default value is not available by way of the `.flox/metadata.json` project state file.

## Checks

- [x] All tests pass.
- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] I have accepted the flox [Contributor License Agreement](../blob/main/.github/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](../blob/main/.github/CONTRIBUTORS.csv) file by way of this pull request or one done previously.

## Release Notes

Updated `flox publish` to optimize the workflow when repeatedly publishing the same package.

### CODE REVIEW

File: pkgs/flox-experimental/default.nix
Comment (ysndr) [inline pkgs/flox-experimental/default.nix:1]: you might want to rename these back without the experimental bit

**Review by ysndr — CHANGES_REQUESTED** [review]

File: flox-bash/lib/commands.sh [hunk 1]
```diff
@@ -108,6 +108,43 @@ function parseNixArgs() {
 	done
 }
 
+# This function parses out common flake options used by the flox
```
Comment (mkenigs) [inline flox-bash/lib/commands.sh:111]: Should 84b83d2dfa8486ea2caf62cd5a418d2199d6eec8 be in a separate PR?

**Review by mkenigs — CHANGES_REQUESTED** [review]

File: flox-bash/lib/commands.sh (hunk 1 above)
Comment (limeytexan) [reply flox-bash/lib/commands.sh:111]: It could be merged separately but it has to come first because it's not possible to publish from anything but the default branch for a repo without it.

Comment (ysndr) [reply flox-bash/lib/commands.sh:111]: Given the description and name of the PR I wouldn't expect this change in this PR..

Aside, I still don't think we should incorporate all the nix flags into the flox cli and we should

**Comment by stahnma** [comment]
Who is working this? It has no assignee.

**Comment by stahnma** [comment]
Who is working this? It has no assignee.

**Comment by ysndr** [comment]
I think @limeytexan is working on this but we currently have two PRs for the same domain and should iron out how to apply this and @aakropotkin's #114.

**Comment by robinbrantley** [comment]
I believe Alex and Michael are working on sorting this out when they meet up today...

File: flox-bash/lib/commands/publish.sh [hunk 2]
```diff
@@ -442,29 +451,8 @@ function floxPublish() {
...
-		keyFile="$FLOX_CONFIG_HOME/secret-key"
-	fi
-	if [ -n "$keyFile" ]; then
-		if [ -f "$keyFile" ]; then
-			$invoke_nix "${_nixArgs[@]}" store sign -r --key-file "$keyFile" $outpaths
-		else
-			error "could not read $keyFile: $!" < /dev/null
-		fi
-	fi
+	floxBuild "${_nixArgs[@]}" --no-link "$canonicalFlakeURL" "${buildArgs[@]}" || \
```
[the remark this answers is not in the document]
Comment (limeytexan) [reply flox-bash/lib/commands/publish.sh:454]: That's a good shout; I tested and you're right that in combination with `--print-out-paths` seems to replace the requirement to read those paths from the eval data. I've gone with your suggestion as I pushed updates to #114 to incorporate these changes into that PR. I'll close out this PR now.
